### ISSUE TITLE
java.util.LinkedHashSet support

### ISSUE THREAD

**Issue by slavarosin** [redisson/redisson#661]
Hello!  

Will it be a big effort to support java.util.LinkedHashSet-like behavior in one of redisson data structures?  

Basically what I need is a collection with element uniqueness and preserved insertion order.  

Would be happy to contribute as well, but I would need some guidance or example where to start from.

**Comment by mrniko** [comment]
Hi!

Redis map keeps insertion order. So you can try to make it like this:
```java
Map<V, Boolean> map = redisson.getMap("mymap");
Set<V> linkedSet = Collections.newSetFromMap(map);
```

**Comment by mrniko** [comment]
You can contribute a new method like `RedissonClient.getLinkedSet` with logic illustrated in previous comment. Don't forget there are two variants of method with and without codec param.

**Comment by slavarosin** [comment]
@mrniko I'm on it already, but what is the proper way to get RSet out of RedissonMap (or Map)? Would like to keep RSet on an interface level of RedissonClient.

**Comment by mrniko** [comment]
> 'm on it already, but what is the proper way to get RSet out of RedissonMap (or Map)?

You're right a new object RedissonLinkedSet needed. It should be a delegator to RedissonMap field.
